src/libcore/panicking.rs

```diff
@@ -48,16 +48,24 @@ pub fn panic(expr_file_line_col: &(&'static str, &'static str, u32, u32)) -> ! {
     // truncation and padding (even though none is used here). Using
     // Arguments::new_v1 may allow the compiler to omit Formatter::pad from the
     // output binary, saving up to a few kilobytes.
-    let (expr, file, line, col) = *expr_file_line_col;
-    panic_fmt(fmt::Arguments::new_v1(&[expr], &[]), &(file, line, col))
+    if cfg!(feature = "panic_immediate_abort") {
+        unsafe { super::intrinsics::abort() }
+    } else {
+        let (expr, file, line, col) = *expr_file_line_col;
+        panic_fmt(fmt::Arguments::new_v1(&[expr], &[]), &(file, line, col))
+    }
 }
 
 #[cold] #[inline(never)]
 #[lang = "panic_bounds_check"]
 fn panic_bounds_check(file_line_col: &(&'static str, u32, u32),
                      index: usize, len: usize) -> ! {
-    panic_fmt(format_args!("index out of bounds: the len is {} but the index is {}",
-                           len, index), file_line_col)
+    if cfg!(feature = "panic_immediate_abort") {
+        unsafe { super::intrinsics::abort() }
+    } else {
+        panic_fmt(format_args!("index out of bounds: the len is {} but the index is {}",
+                               len, index), file_line_col)
+    }
 }
 
 #[cold] #[inline(never)]
```

Comment:
Since this patch (I think?) is primarily about code size implications it's probably worthwhile to make these attributes conditional as well (in addition to the functions above). In `panic_immediate_abort` mode the `inline(never)` isn't necessary and can run the risk of making code size larger because a call instruction is likely more bytes than an abort instruction

Reply:
is `cfg_attr(not(feature="panic_immediate_abort"), inline(never)` the proper way is this case? Or there should be also some `inline(always)`?

Reply:
Yeah that `cfg_attr` should work, and w/ my comment below I think `inline` could probably for suffice for when the feature is turned on?